spring-test/src/main/java/org/springframework/test/context/support/DynamicPropertiesContextCustomizer.java

```diff
@@ -52,6 +51,7 @@
 
 
 	DynamicPropertiesContextCustomizer(Set<Method> methods) {
+		Assert.notEmpty(methods, "'methods' must not be null or empty");
```

Comment:
Since this is a package private class and we know that this constructor is only ever invoked by `org.springframework.test.context.support.DynamicPropertiesContextCustomizerFactory.createContextCustomizer(Class<?>, List<ContextConfigurationAttributes>)`, this assertion does not add much value.

Furthermore, `DynamicPropertiesContextCustomizerFactory` will never invoke this constructor with a null or empty set of methods. If something were to invoke the constructor with `null`, that would be in violation of the package-level `@NonNullApi` declaration, and supplying an empty set would result in the registration of a `DynamicValuesPropertySource` that is effectively a no-op.

In light of that, please revert these changes in this PR.

Reply:
I get it, and thank you for your pointing, I have already reverted these changes.